renku/core/management/datasets.py

```diff
@@ -1209,6 +1211,99 @@ def move_files(self, source_dataset, destination_dataset, files):
         source_dataset.to_yaml()
         destination_dataset.to_yaml()
 
+        if staged_files:
+            files = {
+                str(s.relative_to(self.path)): str(d.relative_to(self.path))
+                for s, d in files
+            }
+            self._replace_paths_in_workflows(files, commit=commit)
+
+    def _replace_paths_in_workflows(self, files, commit):
+        """Replace file paths in dataflows after moving files."""
+
+        def replace_entity_path(entity):
+            replaced = False
+            for entity in entity.entities:
+                if entity.path in files:
+                    replaced = True
+                    entity.replace_path(path=files[entity.path], commit=commit)
+
+            return replaced
+
+        processed = set()
+
+        def replace_paths_in_run(run):
+            replaced = False
+
+            if run._id in processed:
+                return False
+            processed.add(run._id)
+
+            for output in run.outputs or []:
+                entity = output.produces
+                replaced |= replace_entity_path(entity)
+
+            for input_ in run.inputs or []:
+                entity = input_.consumes
+                replaced |= replace_entity_path(entity)
+
+            return replaced
+
+        def replace_paths_in_process(process):
+            replaced = False
+
+            if process._id in processed:
+                return False
+            processed.add(process._id)
+
+            for usage in process.qualified_usage or []:
+                entity = usage.entity
+                replaced |= replace_entity_path(entity)
+
+            for generation in process.generated or []:
+                entity = generation.entity
+                replaced |= replace_entity_path(entity)
+
+            subprocesses = getattr(process, '_processes', [])
+            for p in subprocesses:
+                replaced |= replace_paths_in_process(p)
+
+            if not process.association or not process.association.plan:
+                return replaced
+
+            plan = process.association.plan
+
+            for output in plan.outputs or []:
+                entity = output.produces
+                replaced |= replace_entity_path(entity)
+
+            for input_ in plan.inputs or []:
+                entity = input_.consumes
+                replaced |= replace_entity_path(entity)
+
+            subprocesses = getattr(plan, 'subprocesses', [])
+            for p in subprocesses:
+                replaced |= replace_paths_in_run(p)
+
+            return replaced
+
+        for path in self.workflow_path.glob('*.yaml'):
+            workflow = WorkflowRun.from_yaml(path, client=self)
+
+            replaced = replace_paths_in_process(workflow)
+
+            if replaced:
+                workflow.to_yaml(path=path)
```

Comment:
you need to add the new paths to the activity index https://github.com/SwissDataScienceCenter/renku-python/blob/master/renku/core/management/repository.py#L486 which is used to find workflows that were involved in a path, and probably remove the old path entries from there as well.